src/components/common/Input.interface.ts

```diff
@@ -0,0 +1,28 @@
+export interface Variant {
+  variant: 'filled' | 'outlined' | 'borderless';
+}
+
+export interface AdditionalProps {
+  type?: React.HTMLInputTypeAttribute;
+  error?: {
+    validate: boolean;
+    message: string;
+  };
+  disabled?: boolean;
+  hidden?: boolean;
+  helperText?: string;
+  label?: string;
+  withButton?: boolean;
+  maxLength?: number;
+  withLetterCounting?: boolean;
+  withResetButton?: boolean;
+  prefix?: string;
+  options?: InputHTMLAttributes<HTMLInputElement>;
+  onClick?: () => void;
+  renderStartIcon?: () => JSX.Element;
+  renderEndIcon?: () => JSX.Element;
+  renderButton?: () => ReactNode;
+  onStartIconClick?: () => void;
+  onEndIconClick?: () => void;
+  onReset?: () => void;
+}
```

Comment:
타입스크립트에서 HTMLInput 속성들을 타입으로 정의해놓은게 있는데 해당 부분을 활용하면 더 좋을거 같아요 
`React.InputHTMLAttributes<HTMLInputElement>` 와 `ComponentPropsWithoutRef` 타입 등이 있는데 괜찮은거 활용하시면 좋을거 같아요 

```suggestion
export type Variant = 'filled' | 'outlined' | 'borderless'

export interface AdditionalProps extends React.InputHTMLAttributes<HTMLInputElement> {  error?: {
    validate: boolean;
    message: string;
  };
  helperText?: string;
  withButton?: boolean;
  maxLength?: number;
  withLetterCounting?: boolean;
  withResetButton?: boolean;
  prefix?: string;
  // options?: InputHTMLAttributes<HTMLInputElement>;
  renderStartIcon?: () => JSX.Element;
  renderEndIcon?: () => JSX.Element;
  renderButton?: () => ReactNode;
  onStartIconClick?: () => void;
  onEndIconClick?: () => void;
  onReset?: () => void;
}
```

Reply:
안녕하세요, 승규님!
options로 사용할 생각만 했는데 extends로 하는 더 간단한 방법이 있었네요...
참고해서 수정해보겠습니다!